fedimint-api/src/module/mod.rs

```diff
@@ -411,6 +412,13 @@ pub trait ModuleGen: Debug + Sized {
         task_group: &mut TaskGroup,
     ) -> anyhow::Result<DynServerModule>;
 
+    /// Construct the MigrationMap that contains the migrations from each DatabaseVersion to
+    /// DatabaseVersion + 1. Then apply the migrations to the database atomically.
+    async fn migrate_database(&self, db: Database) -> Result<(), anyhow::Error> {
+        let migrations = MigrationMap::new();
```

Comment:
You probably want `fn ModuleGen::get_database_migrations(&self) -> MigrationMap` and then use it from `fn ModuleGen::init`?

Reply:
Some stuff here is tricky...

Reply:
+1 yes that is better

Reply:
We can play with it over time, no need to stress about it.